yaml.go

```diff
@@ -77,8 +77,19 @@ type Marshaler interface {
 // supported tag options.
 //
 func Unmarshal(in []byte, out interface{}) (err error) {
+	return unmarshal(in, out, false)
+}
+
+// UnmarshalStrict decodes the provided yaml just like Unmarshal but returns
+// type errors also for fields that exists in YAML but are missing from provided
```

Comment:
// UnmarshalStrict is like Unmarshal except that any fields that are found
// in the data that do not have corresponding struct members will result in an error.

?